File: Data Structures & Algorithms/insert-new-interval/submission-3.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        if not len(intervals):
            return [newInterval]
        res = []
        for i in range(len(intervals)):
            interval = intervals[i]
            if interval[1] < newInterval[0]:
                res.append(interval)
            elif interval[0] > newInterval[1]:
                res.append(newInterval)
                res.extend(intervals[i:])
                return res
                break
            else:
                newInterval[1] = max(interval[1], newInterval[1])
                newInterval[0] = min(interval[0], newInterval[0])
        res.append(newInterval)
        return res

File: Data Structures & Algorithms/merge-intervals/submission-3.cs
public class Solution {
    public int[][] Merge(int[][] intervals) {
        var intervalList = intervals.OrderBy(x => x[0]);
        var curInterval = intervalList.First();
        var res = new List<int[]> ();
        foreach (var interval in intervalList)
        {
            if (curInterval[1] < interval[0])
            {
                res.Add(curInterval);
                curInterval = interval;
            }
            else 
            {
                curInterval[1] = Math.Max(interval[1], curInterval[1]);
            }
        }
        res.Add(curInterval);
        return res.ToArray();
    }
}
